homeassistant/components/lametric/config_flow.py

```diff
@@ -245,6 +247,22 @@ async def _async_step_create_entry(self, host: str, api_key: str) -> FlowResult:
             },
         )
 
+    async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
+        """Handle dhcp discovery to update existing entries."""
+        mac = format_mac(discovery_info.macaddress)
+        for entry in self._async_current_entries():
+            if format_mac(entry.data[CONF_MAC]) == mac:
+                self.hass.config_entries.async_update_entry(
+                    entry,
+                    data=entry.data | {CONF_HOST: discovery_info.ip},
+                )
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(entry.entry_id)
+                )
+                return self.async_abort(reason="already_configured")
+
+        return self.async_abort(reason="unknown")
```

Comment:
Should we add the `unknown` abort reason to strings.json?

Reply:
Yeah lets do that to be complete 👍

Reply:
Hmm we don't have a generic translation key for "unknown" when aborting. Should I add it?

Reply:
Usually we use the common error string.